### DISCUSSION TITLE
🧱 Profile Feature Request: Widgets/Blocks

### DISCUSSION THREAD

**Discussion by BlueFalconHD** [community/community#32831] [Other Feature Feedback, Questions, & Ideas]
## 🙋 Request
My request is profile widgets. Imagine the IOS (or android for that matter) home screen grid. Each widget takes up a predefined multiple of space, for example, each space is 100px. So, for a wide widget, it could be 200 wide and 100 tall. This grid would expand dynamically up until a set limit (maybe 10 spaces tall, or 1000px in this case). You would be able to drag widgets around and rearrange just like iOS. The cool part is the way these are defined.

## ⚙️ Configuration
The configuration of these widgets would go something like:

>`@user/.github/widgets.yaml`
```yaml
widgets:
    custom-example:
        name: "Example Widget"
        script: "./widget_scripts/example_widget.js"
        width: 1
        height: 1
```

> `@user/.github/widget_scripts/example_widget.js`
```js
import widget from "widget"; // import the imaginary widget module

let myWidget = new widget.Widget; // define the widget

function getDistance2000 () {
    return (new Date().getFullYear()) - 2000; // a function to get the distance from year 2000
};

myWidget.updateInterval = 86400000; // update the widget content every day

myWidget.reload = () => {
    myWidget.content.displayName = "Years since 2000:"; // set the title at the top left of the widget to inform what the widget means
    myWidget.content.appendText(getDistance2000()); // set the text of the widget to the distance to my birthdate
};

myWidget.register(); // register the widget for processing
```
This results in this widget:
![image](https://user-images.githubusercontent.com/72631767/189777534-b92c8505-4014-4f67-a7c2-a1646d6bf077.png)

## 🚀 Abilities
- Vanilla JavaScript
    - All features of normal JavaScript you can run in the browser.
- Custom text content and graph generation
- Widget API
- GitHub API

**Comment by DenverCoder1** [comment]
Dynamically generated images can fulfill this purpose for the most part.

For example, there are many projects that do this including [github-readme-stats](https://github.com/anuraghazra/github-readme-stats), [github-readme-streak-stats](https://github.com/DenverCoder1/github-readme-streak-stats), [github-readme-youtube-cards](https://github.com/DenverCoder1/github-readme-youtube-cards), [github-readme-activity-graph](https://github.com/Ashutosh00710/github-readme-activity-graph), [spotify-github-profile](https://github.com/kittinan/spotify-github-profile) among many others.

**Reply by BlueFalconHD** [reply]
I am aware of this. But this can waste your monthly github actions along with being a hacky way to fix a big problem. Wanting more profile customization is why those exist in the first place.
